itests/niporep_manual_test.go

```diff
@@ -29,129 +29,347 @@ import (
 			req.NoError(err)
-			t.Logf("MinerA RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
+			t.Logf("genesisMiner RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
 
-			// Miner B should have no power as it has yet to onboard and activate any sectors
-			minerB.AssertNoPower(ctx)
+			// Our test miners should have no power as they have yet to onboard and activate any sectors
+			for _, miner := range miners {
+				miner.AssertNoPower()
+			}
 
-			// Verify that ProveCommitSectorsNI rejects messages with invalid parameters
-			verifyProveCommitSectorsNIErrorConditions(ctx, t, minerB, sealProofType)
-
-			// ---- Miner B onboards a CC sector
-			var bSectorNum abi.SectorNumber
-			var bRespCh chan kit.WindowPostResp
-			var bWdPostCancelF context.CancelFunc
-
-			// Onboard a CC sector with Miner B using NI-PoRep
-			bSectorNum, bRespCh, bWdPostCancelF = minerB.OnboardCCSector(ctx, sealProofType)
-			// Miner B should still not have power as power can only be gained after sector is activated i.e. the first WindowPost is submitted for it
-			minerB.AssertNoPower(ctx)
-
-			// Check that the sector-activated event was emitted
-			{
-				expectedEntries := []types.EventEntry{
-					{Flags: 0x03, Codec: uint64(multicodec.Cbor), Key: "$type", Value: must.One(ipld.Encode(basicnode.NewString("sector-activated"), dagcbor.Encode))},
-					{Flags: 0x03, Codec: uint64(multicodec.Cbor), Key: "sector", Value: must.One(ipld.Encode(basicnode.NewInt(int64(bSectorNum)), dagcbor.Encode))},
-					{Flags: 0x03, Codec: uint64(multicodec.Cbor), Key: "unsealed-cid", Value: must.One(ipld.Encode(datamodel.Null, dagcbor.Encode))},
+			sectors := make([][]kit.OnboardedSector, len(tc.miners))
+
+			for i, tcMiner := range tc.miners {
+				miner := miners[i]
+				var expectSuccesses int
+				for _, good := range tcMiner.goodSectorsToOnboard {
+					if good {
+						expectSuccesses++
+					}
 				}
-				from := head.Height()
-				recentEvents, err := client.FullNode.GetActorEventsRaw(ctx, &types.ActorEventFilter{FromHeight: &from})
+
+				head, err = client.ChainHead(ctx)
 				req.NoError(err)
-				req.Len(recentEvents, 1)
-				req.Equal(expectedEntries, recentEvents[0].Entries)
+
+				// Onboard CC sectors to this test miner using NI-PoRep
+				sectors[i] = miner.OnboardSectors(
+					sealProofType,
+					false,
+					len(tcMiner.goodSectorsToOnboard),
+					kit.WithRequireActivationSuccess(tcMiner.allOrNothing),
+					kit.WithModifySectorsBeforeCommit(func(sectors []*kit.OnboardedSector) error {
+						head, err := client.ChainHead(ctx) // get the latest chain head before onboarding
+						if err != nil {
+							return err
+						}
+						for j, good := range tcMiner.goodSectorsToOnboard {
+							if !good {
+								// Set the seed epoch too far in the past for this sector to be valid
+								sectors[j].SealRandomnessEpoch = head.Height() - 181*builtin.EpochsInDay
```

Comment:
I'm belatedly realising this doesn't quite work for my 6good+1bad real-proofs test; I need to alter the strategy here a little

Reply:
fixed